homeassistant/components/investec/sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Support for balance data via the Investec Bank API."""
+from datetime import timedelta
+import logging
+
+from investec import InvestecClient
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=15)
+
+CONF_ACCOUNTS = "accounts"
+CONF_DATA = "data"
+CONF_ACCOUNT_NAME = "accountName"
+CONF_ACCOUNT_BALANCE = "account_balance"
+CONF_ACCOUNT_ID = "accountId"
+CONF_CURRENCY = "currency"
+CONF_CURRENT_BALANCE = "currentBalance"
+CONF_ACCOUNT_NUMBER = "accountNumber"
+
+ICON = "mdi:cash-multiple"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+    }
+)
```

Comment:
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>